radix-engine-interface/src/blueprints/consensus_manager/invocations.rs

```diff
@@ -70,6 +70,41 @@ impl ConsensusManagerInitialConfiguration {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Default, ScryptoSbor, ManifestSbor)]
+pub struct EpochChangeCondition {
+    /// A lower bound (inclusive) on number of rounds that is assumed to happen during the
+    /// [`duration_millis`].
+    /// If an actual number of rounds after [`duration_millis`] is less than this value, the epoch
+    /// change will wait until this value is reached.
+    pub min_round_count: u64,
+
+    /// An upper bound (exclusive) on number of rounds that is assumed to happen during the
```

Comment:
I'm slightly surprised that the upper bound is exclusive - as it means eg if something goes wrong* we get eg 9999 rounds rather than 10000 rounds but I don't think it matters too much :D.

* like validators slowing down time, which would be a surprising thing to happen.

Reply:
Actually, looks like this is inclusive in the logic? In the sense that once max round is hit, an epoch change occurs.

So perhaps the comment needs updating?

Reply:
Hah, interesting, simply both words "inclusive" and "exclusive" can be used to describe this number :)

In my interpretation, this is an exclusive upper bound of a `range of rounds that we consider normal and under this condition, the timestamp drives the epoch`.
E.g. range `[3, 7)` means that "3, 4, 5 or 6 rounds are totally normal and if this is not exceeded, then timestamp drives the epoch". This implies: "if 7 rounds happen, then the timestamp no longer drives the epoch, but we cut the epoch right away". So 7 is not normal. It is not inside the "expected range of rounds". So it is an exclusive upper bound.

In fact, I wanted to use `expected_round_count: Range<Round>` there, but `Range<>` is not `Sbor` 🤷 

I see that under your definition, we have a "range of round counts that CAN happen in an epoch". Then we could have `[3, 7]` (inclusive) because `7` can be a number of rounds in an epoch (as a special case rule, where we disregard timestamp because of exceeding the normal round limit).

I now rephrased the docs a bit to make it all clear 👍